### DISCUSSION TITLE
Why does bashreadline miss the first event

### DISCUSSION THREAD

**Discussion by iyear** [bpftrace/bpftrace#2821] [Q&A]
After run the `./bashreadline.bt`, it can work fine. But it always misses the first command entered after startup, why is that?

uname: `Linux ubuntu2204 5.15.0-88-generic #98-Ubuntu SMP Mon Oct 2 15:18:56 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux`
bpftrace: `v0.14.0`

**Comment by iyear** [comment]
I may have figured it out. bash was "readline"ing when prompt showed up in my terminal, so the readline wasn't hijacked this time. Is that right?

**Reply by danobi** [reply]
Yeah, that sounds plausible. readline() could've been blocked on IO. Thanks for following up here!
